FeedAPIChallenge/Feed API/RemoteFeedLoader.swift

```diff
@@ -19,6 +19,62 @@ public final class RemoteFeedLoader: FeedLoader {
 	}
 
 	public func load(completion: @escaping (FeedLoader.Result) -> Void) {
-		fatalError("Must be implemented")
+		client.get(from: url) { [weak self] result in
+			guard self != nil else { return }
+			switch result {
+			case let .success((data, response)):
+				completion(FeedLoaderResponseHandler.map(data: data, response: response))
+			case .failure(_):
+				completion(.failure(Error.connectivity))
+			}
+		}
+	}
+}
+
+private final class FeedLoaderResponseHandler {
+	private struct Root: Decodable {
+		private let items: [FeedImageAPIModel]
+
+		var feed: [FeedImage] {
+			items.map { $0.feedImage }
+		}
+	}
+
+	private struct FeedImageAPIModel: Decodable {
+		let id: UUID
+		let description: String?
+		let location: String?
+		let url: URL
+
+		enum CodingKeys: String, CodingKey {
+			case id = "image_id"
+			case description = "image_desc"
+			case location = "image_loc"
+			case url = "image_url"
+		}
+
+		var feedImage: FeedImage {
+			return FeedImage(
+				id: id,
+				description: description,
+				location: location,
+				url: url
+			)
+		}
+	}
+
+	private static let OK_200 = 200
+
+	internal static func map(data: Data, response: HTTPURLResponse) -> FeedLoader.Result {
+		guard response.statusCode == OK_200 else {
+			return .failure(RemoteFeedLoader.Error.invalidData)
+		}
+
+		do {
+			let root = try JSONDecoder().decode(Root.self, from: data)
+			return .success(root.feed)
+		} catch {
+			return .failure(RemoteFeedLoader.Error.invalidData)
+		}
```

Comment:
This expression can be simplified by grouping the branches returning `.failure(RemoteFeedLoader.Error.invalidData)`:

```suggestion
		guard response.statusCode == OK_200,
			  let root = try? JSONDecoder().decode(Root.self, from: data) else {
			return .failure(RemoteFeedLoader.Error.invalidData)
		}
		
		return .success(root.feed)
```